homeassistant/components/overkiz/climate_entities/hitachi_air_to_air_heat_pump_hlrrwifi.py

```diff
@@ -0,0 +1,281 @@
+        """Set new preset mode."""
+        if preset_mode == PRESET_HOLIDAY_MODE:
+            await self._global_control(leave_home=OverkizCommandParam.ON)
+
+        if preset_mode == PRESET_NONE:
+            await self._global_control(leave_home=OverkizCommandParam.OFF)
+
+    def _control_backfill(
+        self, value: str | None, state_name: str, fallback_value: str
+    ) -> str:
+        """Overkiz doesn't accept commands with undefined parameters. This function is guaranteed to return a `str` which is the provided `value` if set, or the current device state if set, or the provided `fallback_value` otherwise."""
+        if value:
+            return value
+        state = self.device.states[state_name]
+        if state and state.value_as_str:
+            return state.value_as_str
+        return fallback_value
+
+    async def _global_control(
+        self,
+        main_operation: str | None = None,
+        target_temperature: int | None = None,
+        fan_mode: str | None = None,
+        hvac_mode: str | None = None,
+        swing_mode: str | None = None,
+        leave_home: str | None = None,
+    ) -> None:
+        """Execute globalControl command with all parameters. There is no option to only set a single parameter, without passing all other values."""
+
+        main_operation = self._control_backfill(
+            main_operation, MAIN_OPERATION_STATE, OverkizCommandParam.ON
+        )
+        target_temperature = target_temperature or self.target_temperature
+
+        fan_mode = self._control_backfill(
+            fan_mode,
+            FAN_SPEED_STATE,
+            OverkizCommandParam.AUTO,
+        )
+        hvac_mode = self._control_backfill(
+            hvac_mode,
+            MODE_CHANGE_STATE,
+            OverkizCommandParam.AUTO,
+        ).lower()  # Overkiz can return states that have uppercase characters which are not accepted back as commands
+        if hvac_mode.replace(
+            " ", ""
+        ) in [  # Overkiz can return states like 'auto cooling' or 'autoHeating' that are not valid commands and need to be converted to 'auto'
+            OverkizCommandParam.AUTOCOOLING,
+            OverkizCommandParam.AUTOHEATING,
+        ]:
+            hvac_mode = OverkizCommandParam.AUTO
+
+        swing_mode = self._control_backfill(
+            swing_mode,
+            SWING_STATE,
+            OverkizCommandParam.STOP,
+        )
+
+        leave_home = self._control_backfill(
+            leave_home,
+            LEAVE_HOME_STATE,
+            OverkizCommandParam.OFF,
+        )
+
+        command_data = [
+            main_operation,  # Main Operation
+            target_temperature,  # Target Temperature
+            fan_mode,  # Fan Mode
+            hvac_mode,  # Mode
+            swing_mode,  # Swing Mode
+            leave_home,  # Leave Home
+        ]
+
+        await self.executor.async_execute_command(
+            OverkizCommand.GLOBAL_CONTROL, *command_data
```

Comment:
```suggestion
            OverkizCommand.GLOBAL_CONTROL, command_data
```

Have you tried if you can just pass the `command_data` as a list? Should be supported as well and the preferred way of doing this.

Reply:
Yes I have tried. But the call fails if I pass the array directly.
```
  File "/home/vscode/.local/lib/python3.11/site-packages/pyoverkiz/client.py", line 977, in check_response
    raise OverkizException(result)
pyoverkiz.exceptions.OverkizException: {'errorCode': 'INCOMPATIBLE_VALUE', 'error': 'globalControl() : Invalid value for command parameter p1 : expected string value (off, on) but got [on, 23, auto, heating, both, off] (ArrayList)'}
```